### PR TITLE
Set up CI with Azure Pipelines

### PR DESCRIPTION
**Pull request by goagain**
This PR is for Azure DevOps CG pipeline. It will do same things with the classic pipeline.

CG pipeline will be auto injected. What we need to do is only run NuGet restore

### CODE REVIEW

File: .github/workflows/azure-devops-pipelines-cg.yml [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+    - main
+  batch: True
+schedules:
+- cron: 0 2 * * 1,2,3,4,5
+  branches:
+    include:
+    - main
+ 
+name: $(date:yyyyMMdd)$(rev:.r)
+resources:
```
Comment (shalinikhare27) [inline .github/workflows/azure-devops-pipelines-cg.yml:21]: Is this necessary here?

Comment (shalinikhare27) [reply .github/workflows/azure-devops-pipelines-cg.yml:21]: The entire `resource` block I mean.

File: .github/workflows/azure-devops-pipelines-cg.yml [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+    type: git
+    ref: refs/heads/main
+
+jobs:
+- job: ComponentGovernance
+  displayName: Component Governance
+  pool:
+    name: Azure Pipelines
+  steps:
+  - checkout: self
```
Comment (shalinikhare27) [inline .github/workflows/azure-devops-pipelines-cg.yml:33]: Again not sure if these are nescessary.

File: .github/workflows/azure-devops-pipelines-cg.yml [hunk 3]
```diff
@@ -0,0 +1,49 @@
+# This pipeline is intended to use inside Azure DevOps for Component Governance,
+# which is may not useful for users outside Microsoft.
+
+variables:
+- name: BuildParameters.solution
+  value: '**\*.sln'
```
Comment (shalinikhare27) [inline .github/workflows/azure-devops-pipelines-cg.yml:6]: You reference `BuildParameters.solution` below in the last step, so why are these defined as variables?

**Review by shalinikhare27 — CHANGES_REQUESTED** [review]

**Review by shalinikhare27 — COMMENTED** [review]
I have a bunch of questions here.

File: .github/workflows/azure-devops-pipelines-cg.yml (hunk 3 above)
Comment (goagain) [reply .github/workflows/azure-devops-pipelines-cg.yml:6]: This yaml is exported from our classic pipeline. Anyway, I can change it.

File: .github/workflows/azure-devops-pipelines-cg.yml (hunk 2 above)
Comment (goagain) [reply .github/workflows/azure-devops-pipelines-cg.yml:33]: I think this part is equivalent to GitHub actions/checkout

File: .github/workflows/azure-devops-pipelines-cg.yml [hunk 4]
```diff
@@ -0,0 +1,49 @@
+# This pipeline is intended to use inside Azure DevOps for Component Governance,
+# which is may not useful for users outside Microsoft.
```
[the remark this answers is not in the document]
Comment (goagain) [reply .github/workflows/azure-devops-pipelines-cg.yml]: Removed

File: .github/workflows/azure-devops-pipelines-cg.yml (hunk 1 above)
Comment (goagain) [reply .github/workflows/azure-devops-pipelines-cg.yml]: Removed. It was auto generated by classic pipeline

**Review by Haard30 — DISMISSED** [review]

**Review by Haard30 — APPROVED** [review]

**Review by shalinikhare27 — APPROVED** [review]

**Pull request merged**